src/s3/commands/presignedURL.ts

```diff
@@ -0,0 +1,61 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { SignedUrlRequest } from '../../shared/clients/s3Client'
+import { Env } from '../../shared/vscode/env'
+import { S3FileNode } from '../explorer/s3FileNode'
+import { Window } from '../../shared/vscode/window'
+import { addCodiconToString } from '../../shared/utilities/textUtilities'
+import { localize } from '../../shared/utilities/vsCodeUtils'
+import { COPY_TO_CLIPBOARD_INFO_TIMEOUT_MS } from '../../shared/constants'
+import { validateTime } from '../util'
+
+export async function presignedURLCommand(
+    node: S3FileNode,
+    window = Window.vscode(),
+    env = Env.vscode()
+): Promise<void> {
+    const time = await promptTime(node.file.key, window)
+    const s3Client = node.s3
+
+    const request: SignedUrlRequest = {
+        bucketName: node.bucket.name,
+        key: node.file.key,
+        operation: 'getObject',
+        time: time,
+    }
+
+    const url = await s3Client.getSignedUrl(request)
+
+    await copyUrl(url, window, env)
```

Comment:
Add error/blank URL handling.

Reply:
With what message could I display the error? Like what would be the reason of this returning nothing or throwing an error?

Reply:
Unknown, but you should always assume that the network calls can error (actually, this might not make a network call?). Some concerns I can identify:
* Credential set does not have `getObject` permissions
* Invalid ARN (in the case of entering an arbitrary ARN if a node isn't provided)